app/services/partnerships/report.rb

```diff
@@ -0,0 +1,78 @@
+# frozen_string_literal: true
+
+module Partnerships
+  class Report
+    CHALLENGE_WINDOW = 14.days.freeze
+    REMINDER_EMAIL_DELAY = 7.days.freeze
+
+    def self.call(**args)
+      new(**args).call
+    end
+
+    def initialize(cohort_id:, school_id:, lead_provider_id:, delivery_partner_id:)
+      @cohort_id = cohort_id
+      @school_id = school_id
+      @lead_provider_id = lead_provider_id
+      @delivery_partner_id = delivery_partner_id
+    end
+
+    def call
+      Partnership.transaction do
```

Comment:
Why not `ActiveRecord::Base.transaction`? We operate on quite a few more things than `Partnership`, will that `transaction` only roll back `Partnership` changes?

Reply:
TBH, not sure why :) There is no real difference, `transaction` is a method defined on `ActiveRecord::Base` class and I'll never understand why class methods are inherited in ruby :P. I **guess** I went for Partnership here because it is a) a bit shorter to type or 2) we are dealing with partnership domain model object. I probably just swap it back to ActiveRecord::Base for consistency.

Or... shall we be using `ApplicationRecord` instead?

Reply:
Ah, technically each model can define its own connection as they can live in a separate dabases (which then would work as a "partial" transaction. In this case I'd say connection on ApplicationRecord is probably the most suitable one... However, it really makes no difference in our case and I don't see any value in doing this, so I'll just change this to ActiveRecord::Base (there's a lot of value in consistency!)